### ISSUE TITLE
CLOUDSQL DB (mysql) running an instance type update

### ISSUE THREAD

**Issue by JarvisIn** [hashicorp/terraform-provider-google#7200]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

$ terraform13 -v
2020/09/04 15:49:19 [INFO] Terraform version: 0.13.1  
2020/09/04 15:49:19 [INFO] Go runtime version: go1.14.7
2020/09/04 15:49:19 [INFO] CLI command args: []string{"version", "-v"}
Terraform v0.13.1
+ provider registry.terraform.io/-/google v3.37.0
+ provider registry.terraform.io/-/google-beta v3.37.0
+ provider registry.terraform.io/-/null v2.1.2
+ provider registry.terraform.io/-/random v2.3.0
+ provider registry.terraform.io/-/template v2.1.2
+ provider registry.terraform.io/hashicorp/google v3.15.0
+ provider registry.terraform.io/hashicorp/google-beta v3.37.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
+ provider registry.terraform.io/hashicorp/random v2.3.0
+ provider registry.terraform.io/hashicorp/template v2.1.2

### Affected Resource(s)

* google_sql_database_instance

### Terraform Configuration Files

```tf
Using the code available here :  https://github.com/gruntwork-io/terraform-google-sql/blob/master/modules/cloud-sql/main.tf
```

### Debug Output

module.mysql["cloudsql-mcs-REDACTED"].google_sql_database_instance.failover_replica[0]: Still modifying... [id=cloudsql-mcs-REDACTED-failover, 5m30s elapsed]
2020/09/04 15:02:45 [ERROR] eval: *terraform.EvalApplyPost, err: Error, failed to update instance settings for : googleapi: Error 412: Condition does not match., staleData
2020/09/04 15:02:45 [ERROR] eval: *terraform.EvalSequence, err: Error, failed to update instance settings for : googleapi: Error 412: Condition does not match., staleData

Error: Error, failed to update instance settings for : googleapi: Error 412: Condition does not match., staleData

  on modules/cloud-sql/main.tf line 160, in resource "google_sql_database_instance" "failover_replica":
 160: resource "google_sql_database_instance" "failover_replica" {

### Expected Behavior

Complete the job with the output processing. 

### Actual Behavior
after creation of master and replica using a configuration based on mysql 5.7, I need to update  the instance type : "db-n1-standard-1" => "db-n1-standard-2".

Running a "terraform apply" : 
- instance type is replaced on the master mysql DB
- instance type is then replaced on the failover database ( replica)

Once the job done, I'm getting the error mentioned above.

If I run a new  "terraform apply", then I get :  Plan: 0 to add, 0 to change, 0 to destroy.
and then the output is processed successfully.

### Steps to Reproduce

Use case :
- create a cloudSQL mysql database with replica
- then change the instance type

NOTA: if there is no replica, things are going well - changing the instance type on the master DB is well processed.

### References

* #0000

**Comment by venkykuberan** [comment]
I am not able to repro it, HTTP 412 is pre-condition not met. Its possible that Master may be up & running while replica is setting up, that's why rerunning the apply works fine.
 Are you getting the error consistently ? If so, please send us the debug log.

**Comment by JarvisIn** [comment]
Hi, 

Yes, the problem appears each time the instance type is updated ...

Actually, the deployment of replicas is working well - this issue is the change related to the instance type, once you got the replica running.

I investigated and discovered when Google is replacing the instance type on the master , the job on the replica instance is done too - both instance types are in sync then.

My understanding : when terraform wants to update the instance type on the replica, things changed - the target instance type is already up to date . This explains as well why the problem appears only when I implement a replica.

So based on this view, I decided to ignore changes done on the instance type defined in the replica resource since Google is doing it, anyway.

I succeeded to fix this issue by adding this block in the definition of resource "google_sql_database_instance" "failover_replica"  :
  lifecycle {
    ignore_changes = [
      settings[0].tier
    ]
  }
